protocol_connect.go

```diff
@@ -559,6 +559,10 @@ func (cc *connectUnaryClientConn) validateResponse(response *http.Response) *Err
 				errors.New(response.Status),
 			)
 		}
+		if wireErr.Code == 0 {
+			// code not set? default to one implied by HTTP status
+			wireErr.Code = connectHTTPToCode(response.StatusCode)
+		}
```

Comment:
Can you also open a PR to add this nuance to the wire protocol docs? This feels less like a behavioral specification and more like an aspect of the wire protocol itself.

Reply:
Roger that.

Reply:
I was in the process of adding this and then read this part of the existing spec, describing the status code of the _Unary-Response_ production:

> When reading data from the wire, client implementations must use the [HTTP-to-Connect mapping](https://connectrpc.com/docs/protocol/#http-to-error-code) to infer a Connect error code if **Bare-Message** is missing or malformed.

IMO, this falls into this category. So it's arguable that always defaulting to "unknown" (as was previously done) was actually a bug.

Reply:
So do you think the existing language needs an update or clarification?